code/game/turfs/turf.dm

```diff
@@ -103,6 +107,29 @@
 /turf/proc/Initalize_Atmos(times_fired)
 	CALCULATE_ADJACENT_TURFS(src)
 
+/turf/examine(mob/user)
+	. += ..()
+	if(!damage_amount)
+		. += "<span class='notice'>It looks fully intact.</span>"
+	else
+		var/dam = round((turf_heat_resistance/turf_max_heat_resistance)* 100, 1)
+		. += "<span class='warning'>The remaining integrity is at [dam]%</span>"
+
+/turf/welder_act(mob/living/user, obj/item/I)
+	. = ..()
+	if(user.a_intent == INTENT_GRAB)
+		if(turf_heat_resistance < turf_max_heat_resistance)
+			if(!I.tool_start_check(user, amount=0))
+				return
+
+			to_chat(user, "<span class='notice'>You begin repairing [src]...</span>")
+			if(I.use_tool(src, user, 40, volume=50))
+				turf_heat_resistance = turf_max_heat_resistance
+				to_chat(user, "<span class='notice'>You repair [src].</span>")
+		else
+			to_chat(user, "<span class='warning'>[src] is already in good condition!</span>")
+		return
```

Comment:
Use early returns and don't add a `return` at the end of the proc. That's already implied.
Use `4 SECONDS` instead of `40` for time units.